esp8266/modules/webrepl.py

```diff
@@ -75,3 +70,51 @@ def start_foreground(port=8266):
     stop()
     s = setup_conn(port, None)
     accept_conn(s)
+
+def _sanitize_password(password):
+    if not password:
+            raise ValueError("Password is required")
+    # Check length.
+    if not 4 <= len(password) <= 8:
+        raise ValueError("Password has to be between 4 and 8 characters long")
+    # Only allow printable ASCII characters.
+    for c in password:
+        if not 32 <= ord(c) <= 126:
+            raise ValueError("Invalid character in the password")
+    # Escape backslashes and quotes.
+    password = password.replace('\\', '\\\\').replace("'", "\\'")
+    return password
+
+def _change_daemon(enable, password=''):
+    LINE_START = "import webrepl; webrepl.start(password='"
```

Comment:
It won't fail. It won't touch any modified user code, which is by design. User data (and code) is sacred.

If the user is advanced enough to modify those lines themselves, they are advanced enough to not rely on this script.

Also, see my note in the comment about moving this whole thing into a separate python module, as a saner solution to this problem.